nibabel/testing/__init__.py

```diff
@@ -62,6 +65,22 @@ def assert_allclose_safely(a, b, match_nans=True, rtol=1e-5, atol=1e-8):
     assert_true(np.allclose(a, b, rtol=rtol, atol=atol))
 
 
+def check_iteration(iterable):
+    """ Checks that an object can be iterated through without errors. """
+    try:
+        for _ in iterable:
+            pass
+    except:
+        return False
+
+    return True
+
+
+def assert_arrays_equal(arrays1, arrays2):
+    for arr1, arr2 in zip_longest(arrays1, arrays2, fillvalue=None):
```

Comment:
I will change it for the last thing you suggested. Even though the iterables of arrays that are being compared are not really big, I prefer using the for loop as I find it more explicit and in the case of `LazyTractogram` objects that avoid calling the `len` function that raises a warning (the one saying it needs to iterate through the streamlines once to get the length).